generators/base/api.d.mts

```diff
@@ -8,16 +9,41 @@ export type ApplicationWithConfig = {
   entities: Record<string, unknown>;
 };
 
-export interface JHipsterGeneratorOptions extends GeneratorOptions {
+export type JHipsterGeneratorOptions = BaseOptions & {
   applicationWithConfig?: ApplicationWithConfig;
   positionalArguments: unknown[];
-}
+  jhipsterContext?: any;
+  skipYoResolve?: boolean;
+  ignoreErrors?: boolean;
+  commandName: string;
+  applicationWithEntities?: any;
+  blueprints?: string;
+  blueprint?: any;
+  configOptions: any;
+  reproducible?: boolean;
+  applicationId?: string;
+  sharedData: any;
+  ignoreNeedlesError?: boolean;
+  skipPriorities?: string[];
+  skipWriting?: boolean;
+  entities?: string[];
+  localBlueprint?: boolean;
+  baseName?: string;
+  db?: string;
```

Comment:
Should it be part of base?

Reply:
Ideally we should build the options type based on the commands definitions, I plan to do it in the future.